commands/bootstrap

```diff
@@ -725,6 +733,28 @@ configure_mtls_mqtt_client() {
     "${EXEC_CMD[@]}" "$TARGET" $SUDO tedge config set mqtt.client.auth.key_file "$LOCAL_CERT_KEY"
 }
 
+c8y_bulk_registration() {
```

Comment:
If I understand correctly, the point here is the register the device from the *operator laptop*. This assumes the laptop is connected to the cloud and to the device - while to the device has no access to the same tenant. This sounds a bit specific. Isn't it?

Also for my understanding: why *bulk* while only one device is registered?

Reply:
> If I understand correctly, the point here is the register the device from the operator laptop. This assumes the laptop is connected to the cloud and to the device - while to the device has no access to the same tenant. This sounds a bit specific. Isn't it?

Depends. The bootstrap script already requires the user's laptop to have internet connectivity as it needs to check if the ca cert has already been uploaded to Cumulocity IoT...so the script is not doing more than it did before.

Though I do see your point if both the device and the user's laptop are not connected to the internet, then the whole thing won't work. This scenario does not seem so uncommon if your factory devices only have 1 network card, and that network is only for the factory environment (which does not have internet), and the user's laptop can be used to bridge between both networks (as generally they can connect to wifi and a local LAN).

I would propose to address the second "both user's laptop and device are offline" in a separate PR.

> Also for my understanding: why bulk while only one device is registered?

The bulk operation API is only for the user related to the certificate. It is not used for child device creation.